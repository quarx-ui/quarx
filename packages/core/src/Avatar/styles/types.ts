import { PickSxBorderSize, SxSize } from '@core/enums';

export type AvatarSize = Exclude<SxSize, 'base'>
export type AvatarBorderRadius = PickSxBorderSize<'rounded' | 'smooth' | 'square'>;

export interface AvatarStyleParams {
    size: AvatarSize,
    radius: AvatarBorderRadius,
    showBorder: boolean,
}
