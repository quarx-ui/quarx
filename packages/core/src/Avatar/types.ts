import { ComponentPropsWithHTML, WithClassesAndStyles } from '@core';
import { ReactNode } from 'react';
import { AvatarStyleKeys, AvatarStyleParams } from './styles';

export interface AvatarPropsWithoutHtml extends
    Partial<AvatarStyleParams>,
    WithClassesAndStyles<AvatarStyleKeys, AvatarStyleParams>,
    AvatarImageProps
{
    className?: string,
    children?: ReactNode,
}

export type AvatarImagePropKeys =
| 'alt'
| 'src'
| 'srcSet'
| 'crossOrigin'
| 'referrerPolicy'

export type AvatarImageProps = Pick<JSX.IntrinsicElements['img'], AvatarImagePropKeys>;
export type AvatarProps = ComponentPropsWithHTML<AvatarPropsWithoutHtml>;
export * from './styles/types';
