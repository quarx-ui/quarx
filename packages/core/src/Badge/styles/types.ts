import { PaletteColor, PickSxBorderSize, PickSxSize } from '@core';

export type BadgeSize = PickSxSize<'small' | 'large'>;
export type BadgeBorderRadius = PickSxBorderSize<'square' | 'smooth' | 'rounded'>;
export type BadgeColor = PaletteColor;
export type BadgeType = 'filled' | 'outline';

export interface BadgeStyleParams {
    /** Размер компонента */
    size: BadgeSize,

    /** Тип заливки компонента.
     * Определяет фон компонента, а также цвет его внутренних элементов
     *
     * @property filled залитый компонент, цвет используемых иконок и фон Counter будет белым (по умолчанию)
     * @property outline фон компонента прозрачный,
     * цвет используемых иконок и фон Counter определяется в соответствии со свойством `color` */
    type: BadgeType,

    /** Цветовое решение компонента.
     * Определяет цвет фона или обводки (в зависимости от выбранного `type`) компонента,
     * а также цвет его внутренних элементов. Значения соответствуют токенам объекта `palette.colors` */
    color: BadgeColor,

    /** Скругление компонента. Значения соответствуют токенам объекта `borderRadii` */
    borderRadius: BadgeBorderRadius,
}
