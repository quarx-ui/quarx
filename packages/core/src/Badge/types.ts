import { ReactNode, Ref } from 'react';
import { CounterProps, PaletteColor, PickSxBorderSize, PickSxSize } from '@core';
import { WithClassesAndStyles } from '@core/styles';
import { BadgeStyleKeys } from './style';

export type BadgeSize = PickSxSize<'small' | 'large'>;
export type BadgeBorderRadius = PickSxBorderSize<'square' | 'smooth' | 'rounded'>;
export type BadgeColor = PaletteColor;
export type BadgeType = 'filled' | 'outline';

export interface BadgeStyleParams {
    /** Размер компонента */
    size?: BadgeSize,

    /** Тип заливки компонента.
     * Определяет фон компонента, а также цвет его внутренних элементов
     *
     * @param filled залитый компонент, цвет используемых иконок и фон Counter будет белым (по умолчанию)
     * @param outline фон компонента прозрачный,
     * цвет используемых иконок и фон Counter определяется в соответствии со свойством `color */
    type?: BadgeType,

    /** Цветовое решение компонента.
     * Определяет цвет фона или обводки (в зависимости от выбранного `type`) компонента,
     * а также цвет его внутренних элементов
     *
     * @param color1 основной светлый цвет
     * @param color2 основной темный цвет
     * @param warning цвет предупреждения
     * @param critical цвет ошибки */
    color?: BadgeColor,

    /** Скругление компонента
     * @param square минимальный радиус скругления и почти острые углы
     * @param smooth среднее скругление
     * @param rounded максимальный радиус скругления, который можно использовать в любом стиле бренда (по умолчанию) */
    borderRadius?: BadgeBorderRadius,
}

export interface BadgeProps extends BadgeStyleParams, WithClassesAndStyles<BadgeStyleKeys, BadgeStyleParams> {
    children: ReactNode,

    /** Элемент, отображаемый с левой стороны компонента */
    leftItem?: ReactNode,

    /** Элемент, отображаемый с правой стороны компонента */
    rightItem?: ReactNode,

    /** Число используемое для отображения во внутреннем компоненте Counter  */
    counter?: number | string,

    /** Объект параметров для настройки внутреннего компонента Counter  */
    counterProps?: Omit<CounterProps, 'children'>,

    /** Ссылка к корневому элементу  */
    ref?: Ref<HTMLDivElement>,

    /** Пользовательский CSS-класс для корневого элемента  */
    className?: string,
}

export type BadgeHtmlAttributes = Omit<JSX.IntrinsicElements['div'], keyof BadgeProps>
