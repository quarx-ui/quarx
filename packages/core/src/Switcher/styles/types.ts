import { Values } from '@core';
import { SWITCHER_POSITION } from '@core/src/Switcher/constants';
import { SelectionSize } from '@core/src/Selection/types';

export type SwitcherPositionType = Values<typeof SWITCHER_POSITION>;

export interface SwitcherStyleParams {
    /** Состояние наведения */
    hover?: boolean,

    /** Размер компонента */
    size?: SelectionSize,

    /** Изменяет цвет компонента уведомляя об ошибке */
    hasError?: boolean,

    /** Изменяет состояние компонента на активное/неактивное */
    disabled?: boolean,

    /** Отключение возможности фокуса */
    disableFocus?: boolean,

    /** Позиция переключателя относительно текста */
    position?: SwitcherPositionType,

    /** Состояние компонента */
    checked?: boolean,
}
