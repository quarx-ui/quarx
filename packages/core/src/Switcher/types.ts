import { ChangeEventHandler, RefObject } from 'react';
import { BaseProps, SelectionProps, Values, WithClassesAndStyles } from '@core';
import { SWITCHER_POSITION } from '@core/src/Switcher/constants';
import { SelectionSize } from '@core/src/Selection/types';
import { SwitcherStyleKeys } from './style';

export type SwitcherPositionType = Values<typeof SWITCHER_POSITION>;

export interface SwitcherStyleParams {
    /** Состояние наведения */
    hover?: boolean,

    /** Размер компонента */
    size?: SelectionSize,

    /** Изменяет цвет компонента уведомляя об ошибке */
    hasError?: boolean,

    /** Изменяет состояние компонента на активное/неактивное */
    disabled?: boolean,

    /** Отключение возможности фокуса */
    disableFocus?: boolean,

    /** Позиция переключателя относительно текста */
    position?: SwitcherPositionType,

    /** Состояние компонента */
    checked?: boolean,
}

export interface SwitcherPropsWithoutHtml extends
    SwitcherStyleParams,
    WithClassesAndStyles<SwitcherStyleKeys, SwitcherStyleParams>,
    BaseProps<HTMLLabelElement>
{
    /** Обработчик изменения состояния элемента `input` */
    onChange?: ChangeEventHandler<HTMLInputElement>,

    /** Свойство `name` элемента `input` */
    name?: string

    /** Свойство `value` элемента `input` */
    value?: string,

    /** Объект со свойствами для элемента `input` */
    inputProps?: JSX.IntrinsicElements['input'],

    /** Ссылка к элементу `input` */
    inputRef?: RefObject<HTMLInputElement>,
}

export type SwitcherPropsHtmlAttributes = Omit<JSX.IntrinsicElements['label'], keyof SwitcherPropsWithoutHtml>;

export type SwitcherProps = SwitcherPropsWithoutHtml & SwitcherPropsHtmlAttributes;

export interface SwitcherSelectionProps extends
    SelectionProps,
    Pick<SwitcherProps, 'checked'>
{
    /** Объект с параметрами для компонента `switcher` */
    switcherProps?: SwitcherProps,
}
