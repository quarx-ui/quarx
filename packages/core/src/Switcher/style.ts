import { colors, SWITCHER_POSITION, SwitcherStyleParams, typography, KeysFromUseStyles, makeStyles } from '@core';
import { paramsToCss } from '@core/utils/paramsToCss';
// TODO: colors from palette

export const useStyles = makeStyles((
    { palette, transitions },
    { size, checked, hasError, disabled, hover, position, disableFocus }: Required<SwitcherStyleParams>,
) => ({
    wrapper: { display: 'inline-flex' },

    root: [
        {
            display: 'flex',
            alignItems: 'center',
            boxSizing: 'border-box',
            cursor: 'pointer',
        },
        disabled && {
            cursor: 'not-allowed',
        },
    ],

    input: [
        {
            margin: 0,
            appearance: 'none',
        },
        !disableFocus && {
            '&:focus-visible + .SxSwitcher-toggleContainer': {
                borderColor: palette.border.focus.dark,
                boxShadow: `inset 0 0 0 1px ${palette.border.focus.dark}`,
                outline: 'none',
            },
        },
    ],

    toggleContainer: [
        {
            display: 'flex',
            alignItems: 'center',
            boxSizing: 'border-box',
            margin: 0,
            padding: 0,
            borderRadius: 10,
            transition: transitions.create(['color', 'background-color', 'border-color', 'box-shadow'], {
                duration: transitions.duration.shortest,
            }),
            border: '1px solid',
            borderColor: colors.Black[24],
            backgroundColor: palette.background.main,
        },
        hover && {
            borderColor: colors.Black[32],
        },
        checked && {
            border: '1px solid',
            borderColor: colors.Emerald[500],
            backgroundColor: colors.Emerald[500],
        },
        checked && hover && {
            border: '1px solid',
            borderColor: colors.Emerald[400],
            backgroundColor: colors.Emerald[400],
        },
        hasError && {
            borderColor: colors.Red[600],
            boxShadow: `inset 0 0 0 1px ${colors.Red[600]}`,
        },
        hasError && checked && {
            backgroundColor: colors.Red[600],
            borderColor: colors.Red[600],
        },
        disabled && {
            borderColor: '#E1E4E5',
            boxShadow: 'none',
            backgroundColor: colors.White[100],
        },
        disabled && checked && {
            border: '1px solid',
            borderColor: '#E1E4E5',
            backgroundColor: '#E1E4E5',
        },
        paramsToCss(size)({
            small: {
                minWidth: 26,
                width: 26,
                height: 16,
            },
            medium: {
                width: 30,
                minWidth: 30,
                height: 18,
            },
            large: {
                minWidth: 34,
                width: 34,
                height: 20,
            },
        }),
    ],

    toggle: [
        {
            marginRight: 2,
            marginLeft: 2,
            borderRadius: 10,
            backgroundColor: colors.Black[24],
            transition: transitions.create(['background-color', 'margin'], {
                duration: transitions.duration.shortest,
            }),
        },
        hover && {
            backgroundColor: colors.Black[32],
        },
        checked && {
            backgroundColor: colors.White[100],
        },
        disabled && {
            backgroundColor: colors.Black[8],
        },
        disabled && checked && {
            backgroundColor: colors.White[100],
        },
        paramsToCss(size)({
            small: {
                width: 10,
                minWidth: 10,
                height: 10,
            },
            medium: {
                width: 12,
                minWidth: 12,
                height: 12,
            },
            large: {
                width: 14,
                minWidth: 14,
                height: 14,
            },
        }),
        checked && paramsToCss(size)({
            small: { marginLeft: 11 },
            medium: { marginLeft: 13 },
            large: { marginLeft: 15 },
        }),
    ],

    content: [
        { color: palette.text.main },
        paramsToCss(size)({
            small: typography.Text.M.Regular,
            medium: typography.Text.L.Regular,
            large: typography.Text.XL.Regular,
        }),
        paramsToCss(position)({
            [SWITCHER_POSITION.left]: { marginLeft: 12 },
            [SWITCHER_POSITION.right]: { marginRight: 12 },
        }),
    ],
}), { name: 'SxSwitcher' });

export type SwitcherStyleKeys = KeysFromUseStyles<typeof useStyles>;
