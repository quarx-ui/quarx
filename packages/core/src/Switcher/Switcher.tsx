/** @jsx jsx */
import { jsx } from '@emotion/react';
import React, { FC, forwardRef, MouseEventHandler, useState } from 'react';
import { SWITCHER_POSITION, SX_SIZE, usePropsOverwrites } from '@core';
import { SwitcherProps } from './types';
import { useStyles } from './style';

export const Switcher: FC<SwitcherProps> = forwardRef<HTMLLabelElement, SwitcherProps>((
    initialProps,
    ref,
) => {
    const { props, cn } = usePropsOverwrites('Switcher', initialProps);
    const {
        styles: externalStyles,
        checked = false,
        size = SX_SIZE.medium,
        hasError = false,
        disabled = false,
        hidden = false,
        hover: externalHover,
        disableFocus = false,
        children,
        position = SWITCHER_POSITION.left,
        inputProps,
        inputRef,
        name,
        value,
        onChange,
        onClick,
        onMouseLeave,
        onMouseEnter,
        ...restProps
    } = props;

    const [hover, setHover] = useState(externalHover ?? false);

    const params = {
        size,
        checked,
        hasError,
        disabled,
        disableFocus,
        hover: externalHover ?? hover,
        position,
    };
    const styles = useStyles({ ...params, styles: externalStyles });

    if (hidden) { return null; }

    const mouseEnterHandler: MouseEventHandler<HTMLLabelElement> = (e) => {
        setHover(true);
        onMouseEnter?.(e);
    };

    const mouseLeaveHandler: MouseEventHandler<HTMLLabelElement> = (e) => {
        setHover(false);
        onMouseLeave?.(e);
    };

    return (
        <label
            ref={ref}
            className={cn('root', params)}
            css={styles.root}
            onMouseEnter={mouseEnterHandler}
            onMouseLeave={mouseLeaveHandler}
            onClick={(e) => {
                if (disabled) { return; }
                onClick?.(e);
            }}
            {...restProps}
        >
            {position === SWITCHER_POSITION.right && children && (
                <div
                    className={cn('content')}
                    css={styles.content}
                >
                    {children}
                </div>
            )}
            <input
                ref={inputRef}
                className={cn('input')}
                css={styles.input}
                type="checkbox"
                name={name}
                value={value}
                disabled={disabled}
                checked={checked}
                onChange={onChange}
                {...inputProps}
            />
            <div
                className={cn('toggleContainer')}
                css={styles.toggleContainer}
            >
                <div
                    className={cn('toggle')}
                    css={styles.toggle}
                />
            </div>
            {position === SWITCHER_POSITION.left && children && (
                <div
                    className={cn('content')}
                    css={styles.content}
                >
                    {children}
                </div>
            )}
        </label>
    );
});
