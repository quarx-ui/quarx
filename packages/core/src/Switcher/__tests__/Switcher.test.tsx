import { cleanup, render, screen } from '@testing-library/react';
import { SwitcherStyleParams } from '@core/src/Switcher/types';
import React, { FC } from 'react';
import { Switcher, SWITCHER_POSITION } from '@core';
import userEvent from '@testing-library/user-event';
import { expectPropsMapInClasses } from '@core/test-utils';

const checkPropsInClasses = (props: Partial<SwitcherStyleParams>) => {
    const {
        size = 'medium',
        hasError = false,
        disabled = false,
        position = SWITCHER_POSITION.left,
    } = props;
    const checkbox = document.querySelector('label');
    if (!checkbox) { return; }
    const propsWithDefault = { size, disabled, hasError, position };

    expectPropsMapInClasses(checkbox as HTMLElement)(propsWithDefault);
};

const checkProps = (Component: FC) => (checkedProps?: Partial<SwitcherStyleParams>) => () => {
    const { asFragment } = render(<Component {...checkedProps}>{Component.displayName}</Component>);

    checkPropsInClasses(checkedProps ?? {});
    expect(asFragment()).toMatchSnapshot();
};

describe('Switcher snapshots', () => {
    Switcher.displayName = 'Switcher';
    const checkSwitcherProps = checkProps(Switcher);

    it('default', checkSwitcherProps());

    it('small', checkSwitcherProps({ size: 'small' }));
    it('medium', checkSwitcherProps({ size: 'medium' }));
    it('large', checkSwitcherProps({ size: 'large' }));

    it('right', checkSwitcherProps({ position: SWITCHER_POSITION.right }));
    it('checked', checkSwitcherProps({ checked: true }));
    it('disabled', checkSwitcherProps({ disabled: true }));
    it('hasError', checkSwitcherProps({ hasError: true }));
});

describe('Switcher', () => {
    it('text should be in the document', () => {
        render(<Switcher>Switcher</Switcher>);

        expect(screen.queryByText('Switcher')).toBeInTheDocument();
        const checkbox = document.querySelector('input');
        expect(checkbox).not.toHaveAttribute('disabled');
    });

    it('Switcher should be checked', () => {
        render(<Switcher checked>Switcher</Switcher>);

        const checkbox = document.querySelector('input');
        expect(checkbox?.checked).toEqual(true);
    });

    it('Switcher should have attribute disabled', () => {
        const mockedOnClick = jest.fn();
        render(<Switcher disabled onClick={mockedOnClick}>Switcher</Switcher>);
        const input = document.querySelector('input');
        const checkbox = document.querySelector('label');

        if (!input || !checkbox) { return; }

        expect(input).toHaveAttribute('disabled');
        userEvent.click(checkbox);
        expect(mockedOnClick).not.toBeCalled();
    });

    it('Switcher should be hidden', () => {
        render(<Switcher hidden>Switcher</Switcher>);
        expect(document.querySelector('input')).not.toBeInTheDocument();

        cleanup();

        render(<Switcher permissions={{ hidden: true }}>Switcher</Switcher>);
        expect(document.querySelector('input')).not.toBeInTheDocument();
    });

    it('onClick should be called', () => {
        let checked = false;
        const mockedOnClick = jest.fn(() => {
            checked = !checked;
        });
        render(<Switcher onChange={mockedOnClick}>Switcher</Switcher>);
        const checkbox = document.querySelector('label');

        if (!checkbox) { return; }

        userEvent.click(checkbox);
        expect(mockedOnClick).toBeCalledTimes(1);
        expect(checked).toEqual(true);
    });
});
