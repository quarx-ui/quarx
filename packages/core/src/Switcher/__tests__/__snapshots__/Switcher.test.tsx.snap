// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher snapshots checked 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_checked SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      checked=""
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1ngaasa-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1ptvyo4-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-1l3e5z1-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots default 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1mgcxk5-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1x8frt1-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-1l3e5z1-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots disabled 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_disabled SxSwitcher_position_left css-1k2y62n-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      disabled=""
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-c6essj-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1od9uj4-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-1l3e5z1-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots hasError 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_hasError SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-mt5fuv-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1x8frt1-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-1l3e5z1-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots large 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_large SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1kgkia8-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-lc2zte-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-w4zvls-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots medium 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1mgcxk5-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1x8frt1-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-1l3e5z1-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots right 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_medium SxSwitcher_position_right css-vhioa4-SxSwitcher-root"
  >
    <div
      class="SxSwitcher-content css-62pq4b-SxSwitcher-content"
    >
      Switcher
    </div>
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1mgcxk5-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1x8frt1-SxSwitcher-toggle"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`Switcher snapshots small 1`] = `
<DocumentFragment>
  <label
    class="SxSwitcher SxSwitcher_size_small SxSwitcher_position_left css-vhioa4-SxSwitcher-root"
  >
    <input
      class="SxSwitcher-input css-140mpmz-SxSwitcher-input"
      type="checkbox"
      value=""
    />
    <div
      class="SxSwitcher-toggleContainer css-1uhjeho-SxSwitcher-toggleContainer"
    >
      <div
        class="SxSwitcher-toggle css-1edh9e7-SxSwitcher-toggle"
      />
    </div>
    <div
      class="SxSwitcher-content css-dc0ipp-SxSwitcher-content"
    >
      Switcher
    </div>
  </label>
</DocumentFragment>
`;
